zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/RealmAwareZkClientTestBase.java

```diff
@@ -0,0 +1,164 @@
+package org.apache.helix.zookeeper.impl.client;
```

Comment:
As a general comment. As we took the approach to composition (aka create new class DedicatedZkClient), theoretically, it would be idea to still test the new method even it is a thin wrapper. 

If we choose the approach of modifying the ZKClient, since the code is not change, we can reason that limiting our testing to only the modified part, namely the checkingPath validity part would be enough.

Considering composition is only a thin wrapper, I don't see the necessity to do it right now. However, later if we change the code of DedicatedZkClient implementation more, we should add more test then.

Reply:
Do you think you'd be able to enhance the tests to our `RealmAwareZkClientTestBase.java` after SharedZkClient is complete? @kaisun2000

Reply:
Discussed offline; @kaisun2000  to add more test cases for SharedZkClient (regarding ephemeral node cases).